### PR TITLE
[ticket/15876] MySQL 5.7+ support for Q&A captcha plugin

### PR DESCRIPTION
**Pull request by nunoplopes**
Add MySQL 5.7+ support for Q&A captcha plugin

Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket: https://tracker.phpbb.com/browse/PHPBB3-15876

### CODE REVIEW

File: phpBB/phpbb/captcha/plugins/qa.php [hunk 1]
```diff
@@ -107,8 +107,7 @@ function init($type)
 
 			$sql = 'SELECT q.question_id, q.lang_iso
 				FROM ' . $this->table_captcha_questions . ' q, ' . $this->table_captcha_answers . ' a
-				WHERE q.question_id = a.question_id
-				GROUP BY lang_iso';
+				WHERE q.question_id = a.question_id';
```
[the remark this answers is not in the document]
Comment (nunoplopes) [reply phpBB/phpbb/captcha/plugins/qa.php]: Well, I think `GROUP BY` without an aggregate function doesn't make sense. At most the code could use `ORDER BY` to sort the output by language, but looking at the nearby code, it doesn't really matter. The questions ids simply go into an array.

[the remark this answers is not in the document]
Comment (nunoplopes) [reply phpBB/phpbb/captcha/plugins/qa.php]: Well, I think `GROUP BY` without an aggregate function isn't very useful. At most the code could use `ORDER BY` to sort the output by language, but looking at the nearby code, it doesn't really matter. The questions ids simply go into an array.

[the remark this answers is not in the document]
Comment (marc1706) [reply phpBB/phpbb/captcha/plugins/qa.php]: Fair enough, was just a thought and it's not really that important in this case.

**Comment by marc1706** [comment]
Thanks for your contribution.

**Pull request merged**